traitsui/context_value.py

```diff
@@ -111,7 +111,7 @@ def CVType(type, **metadata):
         type or an instance of the ContextValue class.
     """
     metadata.setdefault("sync_value", "to")
-    return Either(type, InstanceOfContextValue, **metadata)
+    return Union(type, InstanceOfContextValue, **metadata)
```

Comment:
Either would have defaulted to None here, but None may not have been one of the allowed values.  From the docs:
"If a default value is not defined, Union uses the default value from the first trait in its definition, whereas Either uses None.

For example:

`Either(Int(), Float())`

has a default value of None. However None is not one of the allowed values. If the trait is later set to None from a non-None value, a validation error will occur.

If the trait definition is changed to:

`Union(Int(), Float())`

Then the default value will be 0, which is the default value of the first trait.

To keep None as the default, use None as the first item:

`Union(None, Int(), Float())`

With this, None also becomes one of the allowed values."

It seems like there isn't a way to _exactly_ preserve this behavior?  We may not want /need to preserve the behavior?

Reply:
Actually, in this specific case (since my head is in this example), I suspect that the `Union` behaviour is preferable.

But more generally, yes, we need to be careful of changes in behaviour like this.